### ISSUE TITLE
model and training code for the AR variant

### ISSUE THREAD

**Issue by MikeWangWZHL** [LTH14/mar#34]
Thanks for open-sourcing this amazing project! 
I wonder it is possible to also release model and training code for the AR baseline
<img width="1037" alt="image" src="https://github.com/user-attachments/assets/ea76e8e4-3389-4c3c-9e66-0351f4286b6f">

Thank you in advance!

**Comment by LTH14** [comment]
To keep this repo clean, we don't have a plan to release the AR code in this repo. However, it is very easy to reimplement it using the current repo -- almost all hyper-parameters remain the same as MAR. The only difference is the causal attention mask and the teacher-forcing loss.

**Comment by shaochenze** [comment]
"Hi @LTH14, in the AR variant, is it necessary for the attention mechanism within the MAE encoder to be causal? Alternatively, should we consider removing the MAE encoder altogether in this variant?"

**Comment by LTH14** [comment]
In the AR variant, we don't need the MAE encoder. A single decoder is enough (similar to GPT).

**Comment by shaochenze** [comment]
Thanks! Do you double the depth of MAE decoder?

**Comment by LTH14** [comment]
Yes we keep the total number of parameters unchanged
